src/main/java/org/osc/manager/ism/api/IsmAgentApi.java

```
@@ -16,79 +16,75 @@
 
-    private IsmAgentApi(ApplianceManagerConnectorElement mc, VirtualSystemElement vs) throws Exception {
-        // Set mc as class variable as needed
-        this.log.info("Agent API created for Virtual System: " + vs.getName());
+    public IsmAgentApi(ApplianceManagerConnectorElement mc, VirtualSystemElement vs, TransactionControl txControl,
+            EntityManager em) throws Exception {
+        this.txControl = txControl;
+        this.em = em;
+        this.vs = vs;
+        this.api = new IsmDeviceApi(this.vs, this.txControl, this.em);
     }
 
-    public static IsmAgentApi create(ApplianceManagerConnectorElement mc, VirtualSystemElement vs) throws Exception {
-        return new IsmAgentApi(mc, vs);
+    public static IsmAgentApi create(ApplianceManagerConnectorElement mc, VirtualSystemElement vs,
+            TransactionControl txControl, EntityManager em) throws Exception {
+        return new IsmAgentApi(mc, vs, txControl, em);
     }
 
     @Override
     public List<ManagerDeviceMemberStatusElement> getFullStatus(List<DistributedApplianceInstanceElement> list) {
-        return getAgentFullStatus(list);
-    }
-
-    private List<ManagerDeviceMemberStatusElement> getAgentFullStatus(List<DistributedApplianceInstanceElement> list) {
-        // Get basic status
-        ManagerDeviceMemberStatusElement agentElem = null;
         List<ManagerDeviceMemberStatusElement> response = new ArrayList<>();
-        if (list != null){
-            for (DistributedApplianceInstanceElement dai: list){
-                agentElem = getAgentBasicStatus(dai);
-                response.add(agentElem);
+        if (list != null) {
+            for (DistributedApplianceInstanceElement dai : list) {
+                DeviceMemberEntity member = null;
+                try {
+                    member = (DeviceMemberEntity) this.api.findDeviceMemberByName(dai.getName());
+                } catch (Exception e) {
+                    e.printStackTrace();
```

you should log error not print stack trace.

In this case, if the dai is not found we should throw the exception back to OSC.

---

yes will take care.

---

revert the `if` to avoid deep nesting.... if (list == null) { throw illegalargument...} the rest of the code comes after.